components/sup/src/manager/file_watcher.rs

```diff
@@ -2937,29 +2947,49 @@ mod tests {
                      steps: &[Step]) {
             let mut initial_file = tc_initial_file.clone();
             let mut actual_initial_file = setup.watcher.initial_real_file.clone();
+
+            let is_poll_watcher = match &setup.watcher.get_mut_underlying_watcher().real_watcher {
+                SupWatcher::Native(_watcher) => false,
+                SupWatcher::Fallback(_watcher) => true,
+            };
+
             for (step_idx, step) in steps.iter().enumerate() {
                 self.debug_info.push_level();
                 self.debug_info
                     .add(format!("step {}:\n{}", step_idx, dits!(step)));
-                let iterations = self.execute_step_action(&mut setup, &step.action);
-                self.spin_watcher(&mut setup, iterations);
+                let expected_event_count = self.execute_step_action(&mut setup, &step.action);
+                self.spin_watcher(&mut setup, expected_event_count);
+
+                if is_poll_watcher {
+                    thread::sleep(Duration::from_secs(5));
+                };
                 self.test_dirs(&step.dirs, &setup.watcher.paths.dirs);
```

Comment:
based on my buildkite and local tests, we can remove this sleep.

Reply:
Updated